### ISSUE TITLE
[TEST FLAKE] Intermittently seeing - `No such image: localhost:5000/<test-image>`

### ISSUE THREAD

**Issue by aaron-prindle** [GoogleContainerTools/kaniko#3124]
Currently multiple PRs are seeing the below CI/CD test flake error:
```
        integration_test.go:609: Couldn't get details from image reference for (localhost:5000/docker-dockerfile_test_copyadd_chmod): Couldn't get reference to image localhost:5000/docker-dockerfile_test_copyadd_chmod from daemon: Error response from daemon: No such image: localhost:5000/docker-dockerfile_test_copyadd_chmod:latest
```

where the actual image reference is subject to change.  

I believe this is due to some changes on the GitHub action runner VM or VM image.  Currently this is a flake as some PRs are still able to get all green tests while some seem hit by this.  Still root causing this rcurrently 

Previous runs with this flake (this same PR would succeed [green tests] with no changes in later attempts):
https://github.com/GoogleContainerTools/kaniko/actions/runs/8682504136/job/23807123764#step:7:73

**Comment by mschneider82** [comment]
The integration_test did not find the image which was build with `docker` command. The `BuildDockerImage()` call ignored the returned error. I opened a PR to fix this. After that lets see whats wrong with the build on the chmod issue.

**Comment by mschneider82** [comment]
Solved by #3131 Close this

**Comment by aaron-prindle** [comment]
This was related to DOCKER_BUILDIT not being set in some test envMaps.   In the future, we will likely need to update the test envMap for previous and new tests:

https://github.com/GoogleContainerTools/kaniko/pull/3119#issuecomment-2068932322

Closing, will open a new issue to track that work
